streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableMaterializedValueGetterSupplier.java

```diff
@@ -38,21 +38,21 @@
     }
 
     private class KTableMaterializedValueGetter implements KTableValueGetter<K, V> {
-        private KeyValueStore<K, V> store;
+        private TimestampedKeyValueStore<K, V> store;
 
         @SuppressWarnings("unchecked")
         @Override
         public void init(final ProcessorContext context) {
-            store = (KeyValueStore<K, V>) context.getStateStore(storeName);
+            store = (TimestampedKeyValueStore<K, V>) context.getStateStore(storeName);
         }
 
         @Override
         public V get(final K key) {
-            return store.get(key);
+            final ValueAndTimestamp<V> valueAndTimestamp = store.get(key);
+            return valueAndTimestamp == null ? null : valueAndTimestamp.value();
```

Comment:
This logic has come up once or twice ;)

Just a random idea... What do you think about a static method on `ValueAndTimestamp`:
```java
static <V> V getValueOrNull(final ValueAndTimestamp<V> valueAndTimestamp) {
  return valueAndTimestamp == null ? null : valueAndTimestamp.value();
}
```

Then all these other methods could be:
```suggestion
            return getValueOrNull(valueAndTimestamp);
```
?

Reply:
Ack. I'll just update the KIP accordingly. Turn out, we already have `ValueAndTimestamp#make` and no public constructor (also not documented on the KIP).